prime-angular-frontend/src/app/modules/paper-enrolment/shared/guards/finalized-enrolment.guard.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable, Inject } from '@angular/core';
+import {
+  Router,
+  Params,
+  CanActivate,
+  ActivatedRouteSnapshot,
+  RouterStateSnapshot,
+  UrlTree,
+  UrlSegment
+} from '@angular/router';
+import { PaperEnrolmentResource } from '@paper-enrolment/services/paper-enrolment-resource.service';
+import { HttpEnrollee } from '@shared/models/enrolment.model';
+import { PaperEnrolmentRoutes } from '@paper-enrolment/paper-enrolment.routes';
+import { Observable } from 'rxjs';
+import { AppConfig, APP_CONFIG } from 'app/app-config.module';
+import { map } from 'rxjs/operators';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class FinalizedEnrolmentGuard implements CanActivate {
+  constructor(
+    private router: Router,
+    @Inject(APP_CONFIG) private config: AppConfig,
+    private paperEnrolmentResource: PaperEnrolmentResource
+  ) { }
+
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    const url = this.getUrl(state);
+    return this.checkAccess(url, next.params);
+  }
+
+  protected checkAccess(routePath: string = null, params: Params): Observable<boolean> | Promise<boolean> {
+    const enrolleeId = +params.eid;
+    if (enrolleeId) {
+      return this.paperEnrolmentResource.getEnrolleeById(enrolleeId)
+        .pipe(
+          map((enrollee: HttpEnrollee) => {
+            return this.routeDestination(routePath, enrollee, params);
+          })
+        );
+    } else {
+      return new Promise(async (resolve, reject) => resolve(true));
```

Comment:
You can simplify this by dropping the `else` and returning `of(true)`.  Where `of` will emit a list of values then complete so in this case emit `true` and complete.  With respect to the else there's nothing else to run so it's superfluous and can be omitted.

```
if(someStatement) {
  return this.paperEnrolmentResource.getEnrolleeById(...)
    .pipe(...);
}

return of(true);
```